### PR TITLE
[AAE-10390] Storybook Dialog Module

### PR DESCRIPTION
**Pull request by marcol13**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
[https://alfresco.atlassian.net/browse/AAE-10390](https://alfresco.atlassian.net/browse/AAE-10390)

**What is the new behaviour?**
Two new stories for DownloadZip and EditJSON dialogs:

- **Download ZIP Component**

image

- **Edit JSON Component**

image

Video: 

https://user-images.githubusercontent.com/56632321/186692010-9393b903-aa29-4ab8-8268-c7b212898f60.mp4

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/dialogs/download-zip/download-zip.dialog.stories.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+export default {
+    component: DownloadZipDialogStorybookComponent,
+    title: 'Core/Dialog/Download ZIP Dialog',
+    decorators: [
+        moduleMetadata({
+            imports: [
+                CoreStoryModule,
+                DownloadZipDialogModule,
+                MatButtonModule
```
Comment (tomgny) [inline lib/core/dialogs/download-zip/download-zip.dialog.stories.ts:44]: can we move it to the DownloadZipDialogModule?

File: lib/core/dialogs/edit-json/edit-json.dialog.stories.component.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    set settings(newSettings: EditJsonDialogSettings) {
+        this._settings = {
+            title: newSettings.title,
+            editable: newSettings.editable,
+            value: JSON.stringify(newSettings.value, null, '  ')
+        };
+    }
+
+    @Input()
```
Comment (tomgny) [inline lib/core/dialogs/edit-json/edit-json.dialog.stories.component.ts:40]: inputs should be at the top of class body

File: lib/core/dialogs/edit-json/edit-json.dialog.stories.component.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    EditJsonDialogComponent,
+    EditJsonDialogSettings
+} from './edit-json.dialog';
+
+@Component({
+    selector: 'adf-edit-json-dialog-storybook',
+    template: `<button mat-raised-button (click)="openDialog()">Open dialog</button>`
+})
+export class EditJsonDialogStorybookComponent implements OnInit, OnChanges {
+    private _settings: EditJsonDialogSettings;
```
Comment (tomgny) [inline lib/core/dialogs/edit-json/edit-json.dialog.stories.component.ts:30]: why we use underscore here?

File: lib/core/dialogs/edit-json/edit-json.dialog.stories.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    measurementId: 'm_10001',
+    researcherId: 's_10002'
+};
+
+export default {
+    component: EditJsonDialogStorybookComponent,
+    title: 'Core/Dialog/Edit JSON Dialog',
+    decorators: [
+        moduleMetadata({
+            imports: [CoreStoryModule, EditJsonDialogModule, MatButtonModule]
```
Comment (tomgny) [inline lib/core/dialogs/edit-json/edit-json.dialog.stories.ts:37]: the same as above, can we move it to the module level?

File: lib/core/dialogs/download-zip/download-zip.dialog.stories.ts (hunk 1 above)
Comment (marcol13) [reply lib/core/dialogs/download-zip/download-zip.dialog.stories.ts]: In both of stories we use special components built just for Storybook, so MatButtonModule should be here as far as we don't want to put that component into some Module

**Pull request merged**